src/Core__Math.resi

```diff
@@ -0,0 +1,807 @@
+  See [`Math.abs`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/abs) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Constants.sqrt2
+  ```
+  */
+  @val
+  external sqrt2: float = "Math.SQRT2"
+}
+
+/**
+Provide Math utilities for `int`
+*/
+module Int: {
+  /**
+  `abs(v)` returns absolute value of `v`.
+  See [`Math.abs`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/abs) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.abs(-2) // 2
+  Math.Int.abs(3) // 3
+  ```
+  */
+  @val
+  external abs: int => int = "Math.abs"
+
+  /**
+  `clz32(v)` returns the number of leading zero bits of the argument's 32 bit
+  int representation.
+  See [`Math.clz32`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/clz32) on MDN.
+
+  ## Examples
+
+  ```rescript
+  // 00000000000000000000000000000001
+  Math.Int.clz32(1) // 31
+  // 00000000000000000000000000000100
+  Math.Int.clz32(4) // 29
+  ```
+  */
+  @val
+  external clz32: int => int = "Math.clz32"
+
+  /**
+  `imul(a, b)` returns 32-bit integer multiplication. Use this only when you
+  need to optimize performance of multiplication of numbers stored as 32-bit
+  integers.
+  See [`Math.imul`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/imul) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.imul(3, 4) // 12
+  Math.Int.imul(-5, 12) // 60
+  ```
+  */
+  @val external imul: (int, int) => int = "Math.imul"
+
+  /**
+  `min(a, b)` returns the minimum of its two integer arguments.
+  See [`Math.min`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.min(1, 2) // 1
+  Math.Int.min(-1, -2) // -2
+  ```
+  */
+  @val external min: (int, int) => int = "Math.min"
+
+  /**
+  `minMany(arr)` returns the minimum of the integers in the given array `arr`.
+  Returns `Infinity` if `arr` is empty.
+  See [`Math.min`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.minMany([1, 2]) // 1
+  Math.Int.minMany([-1, -2]) // -2
+  Math.Int.minMany([])->Float.isFinite // false
+  ```
+  */
+  @variadic @val external minMany: array<int> => int = "Math.min"
+
+  /**
+  `max(a, b)` returns the maximum of its two integer arguments.
+  See [`Math.max`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.max(1, 2) // 2
+  Math.Int.max(-1, -2) // -1
+  ```
+  */
+  @val external max: (int, int) => int = "Math.max"
+
+  // NOTE: `Math.Int.maxMany([])` compile to `Math.max()` and return `Infinity`
+  // In JS `Math.max([])` return 0
+  // We should create a wrapper to handle an empty array?
+  // Same problem with `Math.Int.minMany`
```

Comment:
Let's open a separate GitHub issue for this perhaps, what do you think? Same for the `pow` stuff below.

Reply:
Ok